### DISCUSSION TITLE
Improvements to Tag functionality

### DISCUSSION THREAD

**Discussion by BrekiTomasson** [stashapp/stash#3068] [Ideas]
We had a long chat on Discord today regarding possible future development ideas for tags, and I took it upon myself to summarize the three salient points here for future reference:

# Date Ranges for Performer Tags

When a tag is added to a performer, one should be able to add an **optional** start date and/or an **optional** end date for that tag. An easy way to do this in the UI would be to hover over the tag while on the performer page, which would reveal the current Tag info popup but with two additional date fields that one can populate with information.

This added metadata could then be used when doing Performer Tag-based searches, so that a tag that is only valid between 2019-2021 will be ignored for scenes with a date in 2022, for example. This could be useful for a number of different types of things, such as performers undergoing plastic surgery, changing hair color, gaining/losing a lot of weight, etc. A given performer tag might only be relevant for that performer during a given date range.

This functionality would be *optional* and near-invisible unless you're actively hovering over the performer tags.

One potential point that might raise the complexity of this implementation is that it would probably mean you would be able to add the same tag more than once to any performer, so long as the date ranges do not overlap. This would have to be solved in a way that makes it intuitive.

# Performer Identification for Scene Tags

Similar to the previous point, when viewing a scene, hovering over a Scene Tag would allow you to check which particular performer(s) that scene tag is valid for (with the list taken from the list of performers attached to that scene). This would allow you to say that the "Schoolgirl" tag is relevant for one of the two female performers only, but that the "Step Dad" tag is relevant for the male performer, for example.

This added metadata could then be used when doing scene searches, allowing you to do searches like "Scenes where Performer A is tagged with 'Schoolgirl'", rather than just "Scenes with Performer A where the scene itself has the "Schoolgirl" tag". 

Again, this functionality would be *optional* and near-invisible unless you're actively hovering over the scene tags.

# Distinct Tag Types

A tag should be categorizable as "Performer Tag Only", "Scene Tag Only" or "Both", with "Both" being the default. This would help a lot when adding tags to performers, as all of the scene tags would not obstruct the view (and vice versa). As performer tags describe the "who" and scene tags describe the "what", "where" and "how", there is little to no overlap between them, making a separation like this fairly logical.

It would also clear up a lot of potential confusion and annoyance in the UI, as the "Outdoor" tag doesn't make sense for a performer and the "Heavily Freckled" tag doesn't make sense for a scene.

Again, this functionality would be *optional* and default usage would work just like the tags work today.

**Comment by stg-annon** [comment]
#1253 related to **Distinct Tag Types**
